sds/src/scanner/mod.rs

```diff
@@ -1130,6 +1131,30 @@ mod test {
         assert_eq!(content, "[GITHUB] ghp_M7H4jxUDDWHP4kZ6A4dxlQYsQIWJuq11T4V5");
     }
 
+    #[test]
+    fn test_nhs_checksum() {
+        let pattern = ".+";
+        let rule = RegexRuleConfig::builder(pattern.to_string())
+            .match_action(MatchAction::Redact {
+                replacement: "[NHS]".to_string(),
+            })
+            .build();
+
+        let rule_with_checksum = RuleConfigBuilder::from(&rule)
+            .validator(NhsCheckDigit)
+            .build();
```

Comment:
Nit: this can be simplify a little bit.
```suggestion
        let rule_with_checksum = RegexRuleConfig::builder(pattern.to_string())
            .match_action(MatchAction::Redact {
                replacement: "[NHS]".to_string(),
            })
            .validator(NhsCheckDigit)
            .build();
```

Reply:
done